--- ./work/tools/build/src/tools/python.jam.orig	2019-09-28 13:13:54.000000000 +0200
+++ ./work/tools/build/src/tools/python.jam	2019-09-28 13:14:43.000000000 +0200
@@ -1017,12 +1017,22 @@
     {
         alias python_for_extensions : python : $(target-requirements) ;
     }
-    else if $(target-os) = darwin {
+    else if $(target-os) = darwin
+    {
+        local cb-sysroot = "//" ;
+        if [ os.environ CONDA_BUILD_SYSROOT ]
+        {
+            cb-sysroot = [ os.environ CONDA_BUILD_SYSROOT ] ;
+        }
+
         alias python_for_extensions
             :
             : $(target-requirements)
             :
-            : $(usage-requirements) <linkflags>"-undefined dynamic_lookup"
+            : $(usage-requirements)
+                <linkflags>"-undefined dynamic_lookup"
+                <linkflags>"-isysroot $(cb-sysroot)"
+                <library-path>$(libraries)
             ;
     }
     # On AIX we need Python extensions and Boost.Python to import symbols from
