From b9fe808ff49ee379c785542a1821c7a9be9bd3fa Mon Sep 17 00:00:00 2001
From: Glen Fernandes <glen.fernandes@gmail.com>
Date: Mon, 17 Apr 2023 06:59:02 -0400
Subject: [PATCH] Define unary_function and binary_function unconditionally

---
 include/boost/functional.hpp | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/include/boost/functional.hpp b/include/boost/functional.hpp
index 6443078..6c63146 100644
--- a/include/boost/functional.hpp
+++ b/include/boost/functional.hpp
@@ -21,7 +21,6 @@ namespace boost
     namespace functional
     {
         namespace detail {
-#if defined(_HAS_AUTO_PTR_ETC) && !_HAS_AUTO_PTR_ETC
             // std::unary_function and std::binary_function were both removed
             // in C++17.
 
@@ -39,12 +38,6 @@ namespace boost
                 typedef Arg2 second_argument_type;
                 typedef Result result_type;
             };
-#else
-            // Use the standard objects when we have them.
-
-            using std::unary_function;
-            using std::binary_function;
-#endif
         }
     }
 
